### DISCUSSION TITLE
Conclusão - Fundamentos da Orientação a Objetos

### DISCUSSION THREAD

**Discussion by Lucas-back-end** [balta-io/forum#520] [Tira Dúvidas] [answered]
Olá! 

Acabei de finalizar o curso, Fundamentos da Orientação a Objetos. Eu entendi perfeitamente os conceitos de todos os tópicos, mas confesso que na hora do, mão na massa, eu entendi as chamadas, mas achei muito difícil de assimilar todas as classes e consequentemente por em prática o conceito OOP, enquanto o professor balta ia ensinando.

**É normal essa fase de dificuldade de assimilação, tendo em vista que estou começando?** 
**O que posso fazer para desenvolver essa assimilação? Não consigo criar nada ainda para poder exercitar. Fico com medo de lá na frente fracassar.**

**Comment by henriqueweege** [comment, marked as answer]
Oi, @Lucas-back-end !

Fica tranquilo, OOP é difícil mesmo e leva algum tempo para entender. Passar por dificuldades faz parte de todo processo de aprendizado, e de maneira alguma quer dizer que você vai fracassar... bom, a menos que desista de aprender 😅😅

O que gosto de fazer quando me sinto dessa maneira é dar tempo ao tempo. As vezes a gente precisa se dar o tempo para assimilar o conteúdo, especialmente com cursos tão ricos e profundos quanto os do Balta. 
Outra coisa que é legal de fazer é reassistir as aulas, e além disso, pesquisar outras fontes que também ensinam sobre o assunto. As vezes só de você olhar o objeto de estudo de um ângulo diferente já vai te ajudar demais.

Da uma olhada [nesse artigo do Balta](https://balta.io/blog/orientacao-a-objetos) e [esse do dev media](https://www.devmedia.com.br/os-4-pilares-da-programacao-orientada-a-objetos/9264). São excelentes e me ajudaram a entender OOP.

Enfim, espero que seja de ajuda. 
Valeeeeu!

**Reply by Lucas-back-end** [reply]
Ope!

Foi de grande ajuda sim! Agradeço imensamente pelo apoio, idéias e referências prestadas.
